language/transaction-builder/src/lib.rs

```diff
@@ -564,3 +569,26 @@ encode_txn_script! {
     script: RemoveParentAccount,
     doc: "Removes the parent account at `parent_address`. It can be recertified in the future however."
 }
+
+//...........................................................................
+// WriteSets
+//...........................................................................
+
+pub fn encode_stdlib_upgrade_transaction(option: StdLibOptions) -> ChangeSet {
+    let mut write_set = WriteSetMut::new(vec![]);
+    let stdlib = stdlib::stdlib_modules(option);
+    for module in stdlib {
+        let mut bytes = vec![];
+        module
+            .serialize(&mut bytes)
+            .expect("Failed to serialize module");
+        write_set.push((
+            AccessPath::code_access_path(&module.self_id()),
```

Comment:
If the on-chain stdlib previously contained a module named `Foo`, but `Foo` is deleted in the local stdlib, will `Foo` still exist on-chain after running this upgrade transaction?

Reply:
I guess there's no way for us to capture that just in the `transaction-builder` right? As there's no way for us to know what the "old" stdlib modules are.

Reply:
Not in `transaction-builder` directly, but perhaps we could read this list from the blockchain in the client + pass it to `transaction-builder` (or do the check there)?